sorting/threeway_merge_sort.cpp

```diff
@@ -0,0 +1,154 @@
+/**
+ *  @{
+ *  @file
+ *  @brief 3-Way Merge Sort Algorithm implementation
+ *
+ *  @author Nirmalya Pal (http://github.com/SpEXterXD)
+ *
+ *  @details
+ *  3-Way Merge Sort is an efficient, general-purpose, comparison
+ *  based sorting algorithm. It extends the traditional Merge Sort
+ *  algorithm by dividing the input array into three parts instead of two.
+ *
+ */
+
+#include <iostream>
```

Comment:
headers should be documented https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines